src/v/kafka/client/client.cc

```diff
@@ -15,8 +15,8 @@
 #include "kafka/client/exceptions.h"
 #include "kafka/client/logger.h"
 #include "kafka/client/partitioners.h"
-#include "kafka/client/retry_with_mitigation.h"
 #include "kafka/client/sasl_client.h"
+#include "kafka/client/utils.h"
```

Comment:
> This commit seems to contain the move of the file

Yes, this commit renames retry_with_mitigation.h to utils.h

> which was mentioned in the previous commit, but not done.

It was not mentioned in the previous commit, so I am confused by your comment.

Reply:
So the commits go:
* Factor out the retry impl into an existing file
* Move that file and add code to find the leader
* Remove existing code for finding a leader

I think it would be clearer to have commits that:
* Rename the file
* Refactors out the impl
* Refactor finding a group leader to use the impl

Anyway, it's not a blocker.